raiden/network/transport/matrix/rtc/web_rtc.py

```diff
@@ -360,99 +390,79 @@ async def reset(self) -> None:
+        hangup_message = {
+            "type": _RTCMessageType.HANGUP.value,
+            "call_id": self.call_id,
+        }
+        self._signaling_send(self.partner_address, json.dumps(hangup_message))
+
+    def handle_sdp_callback(
+        self, rtc_session_description: Optional[RTCSessionDescription]
     ) -> None:
         """
-        This is a callback function to process sdp (session description protocol) messages.
+        This is a callback function to process SDP (Session Description Protocol) messages.
         These messages are part of the ROAP (RTC Offer Answer Protocol) which is also called
-        signalling. Messages are exchanged via the partners' private matrix room.
+        signalling. Messages are exchanged via Matrix.
         Args:
             rtc_session_description: sdp message for the partner
-            partner_address: Address of the partner
         """
         if rtc_session_description is None:
             return
 
-        rtc_partner = self.get_rtc_partner(partner_address)
-
         sdp_type = rtc_session_description.type
         message = {
             "type": sdp_type,
             "sdp": rtc_session_description.sdp,
-            "call_id": rtc_partner.call_id,
+            "call_id": self.call_id,
         }
         self.log.debug(
             f"Send {sdp_type} to partner",
-            partner_address=to_checksum_address(partner_address),
+            partner_address=to_checksum_address(self.partner_address),
             sdp_description=message,
         )
 
-        self._send_signaling_message(partner_address, json.dumps(message))
+        self._signaling_send(self.partner_address, json.dumps(message))
 
-    def _maybe_initialize_web_rtc(self, address: Address) -> None:
-        if address in self._web_rtc_channel_inits:
-            return
 
-        lower_address = my_place_or_yours(self.node_address, address)
+@dataclass
+class WebRTCManager(_CoroutineHandler, Runnable):
+    def __init__(
+        self,
+        node_address: Address,
+        process_messages: Callable[[List[ReceivedRaidenMessage]], None],
+        signaling_send: Callable[[Address, str], None],
+        stop_event: GEvent,
+    ) -> None:
+        super().__init__()
+        self.node_address = node_address
+        self._process_messages = process_messages
+        self._signaling_send = signaling_send
+        self._stop_event = stop_event
+        # the addresses for which we're currently trying to initialize WebRTC channels
+        self._web_rtc_channel_inits: Set[Address] = set()
+        self._address_to_connections: Dict[
+            Address, Dict[str, _RTCConnection]
+        ] = collections.defaultdict(dict)
+        self.log = log.bind(node=to_checksum_address(node_address))
+
+    def _handle_message(self, message_data: str, partner_address: Address) -> None:
+        messages: List[ReceivedRaidenMessage] = []
+        for msg in validate_and_parse_message(message_data, partner_address):
```

Comment:
Doesn't parsing usually include validation? At least Marshmallow and other libraries like that include that step on deserialization as far as I remember.

Would it make sense to have the class `ReceivedRaidenMessage` also do validation and therefore use that to parse and validate instead?

Reply:
> Doesn't parsing usually include validation?

Typically, yes, but it also depends on the kind of validation. E.g. something may be a valid thing, as far as parsing is concerned, but there could be additional validation steps needed (consider, for instance, parsing an integer and checking whether the integer is a square of some other number).

In any case, I just used the existing method here. If there is something to be done, that would be topic for another PR.